packages/dev/core/src/Rendering/renderingManager.ts

```diff
@@ -85,13 +85,41 @@ export class RenderingManager {
     private _customTransparentSortCompareFn: { [id: number]: Nullable<(a: SubMesh, b: SubMesh) => number> } = {};
     private _renderingGroupInfo: Nullable<RenderingGroupInfo> = new RenderingGroupInfo();
 
+    private _maintainStateBetweenFrames = false;
     /**
      * Gets or sets a boolean indicating that the manager will not reset between frames.
      * This means that if a mesh becomes invisible or transparent it will not be visible until this boolean is set to false again.
      * By default, the rendering manager will dispatch all active meshes per frame (moving them to the transparent, opaque or alpha testing lists).
      * By turning this property on, you will accelerate the rendering by keeping all these lists unchanged between frames.
      */
-    public maintainStateBetweenFrames = false;
+    public get maintainStateBetweenFrames() {
+        return this._maintainStateBetweenFrames;
+    }
+
+    public set maintainStateBetweenFrames(value: boolean) {
+        if (value === this._maintainStateBetweenFrames) {
+            return;
+        }
+
+        this._maintainStateBetweenFrames = value;
+
+        // Restore wasDispatched flags when switching to maintainStateBetweenFrames to false
+        if (!this._maintainStateBetweenFrames) {
+            for (let mesh of this._scene.meshes) {
+                for (let subMesh of mesh.subMeshes) {

```

Comment:
To be safe, I think you should check that `mesh.subMeshes` is not null/undefined because even if it is defined as `public subMeshes: SubMesh[];` I think it is possible this property is empty... At least, there are a lot of code in the codebase which is checking `mesh.subMeshes` before dealing with the property.